src/Persistence/Sql.php

```diff
@@ -662,117 +662,52 @@ public function action(Model $model, string $type, array $args = [])
         return $query;
     }
 
-    /**
-     * Tries to load data record, but will not fail if record can't be loaded.
-     *
-     * @param mixed $id
-     */
     public function tryLoad(Model $model, $id): ?array
     {
-        if (!$model->id_field) {
-            throw (new Exception('Unable to load field by "id" when Model->id_field is not defined.'))
-                ->addMoreInfo('id', $id);
-        }
+        $noId = $id === self::ID_LOAD_ONE || $id === self::ID_LOAD_ANY;
 
         $query = $model->action('select');
-        $query->where($model->getField($model->id_field), $id);
-        $query->limit(1);
 
-        // execute action
-        try {
-            $dataRaw = $query->getRow();
-            if ($dataRaw === null) {
-                return null;
+        if (!$noId) {
+            if (!$model->id_field) {
+                throw (new Exception('Unable to load field by "id" when Model->id_field is not defined.'))
+                    ->addMoreInfo('id', $id);
             }
-            $data = $this->typecastLoadRow($model, $dataRaw);
-        } catch (DsqlException $e) {
-            throw (new Exception('Unable to load due to query error', 0, $e))
-                ->addMoreInfo('query', $query->getDebugQuery())
-                ->addMoreInfo('message', $e->getMessage())
-                ->addMoreInfo('model', $model)
-                ->addMoreInfo('scope', $model->scope()->toWords());
-        }
-
-        if (!isset($data[$model->id_field])) {
-            throw (new Exception('Model uses "id_field" but it wasn\'t available in the database'))
-                ->addMoreInfo('model', $model)
-                ->addMoreInfo('id_field', $model->id_field)
-                ->addMoreInfo('id', $id)
-                ->addMoreInfo('data', $data);
+            $query->where($model->getField($model->id_field), $id);
         }
-
-        return $data;
-    }
-
-    /**
-     * Loads a record from model and returns a associative array.
-     *
-     * @param mixed $id
-     */
-    public function load(Model $model, $id): array
-    {
-        $data = $this->tryLoad($model, $id);
-
-        if (!$data) {
-            throw (new Exception('Record was not found', 404))
-                ->addMoreInfo('model', $model)
-                ->addMoreInfo('id', $id)
-                ->addMoreInfo('scope', $model->scope()->toWords());
-        }
-
-        return $data;
-    }
-
-    /**
-     * Tries to load any one record.
-     */
-    public function tryLoadAny(Model $model): ?array
-    {
-        $load = $model->action('select');
-        $load->limit(1);
+        $query->limit($id === self::ID_LOAD_ANY ? 1 : 2);
```

Comment:
smart move with limit(2) :)

Reply:
yes, performance should be fine

> Only question is - what if table will actually have only one record? Then it will not be able to detect issue here.

that is expected and not only if table have not one record, but also if conditions match only one record :)